spec/workers/republish_worker_spec.rb

```diff
@@ -0,0 +1,101 @@
+require "rails_helper"
+
+RSpec.describe RepublishWorker do
+  let(:document) { FactoryGirl.create(:cma_case) }
+  let(:content_id) { document["content_id"] }
+
+  let(:republish_matcher) {
+    request_json_includes("update_type" => "republish")
+  }
+
+  before do
+    stub_any_publishing_api_call
+    stub_any_rummager_post
+
+    publishing_api_has_item(document)
+  end
+
+  %w(draft redrafted).each do |publication_state|
+    context "when the publication_state is '#{publication_state}'" do
+      let(:document) {
+        FactoryGirl.create(:cma_case, publication_state: publication_state)
+      }
+
+      it "sends the document to the publishing api" do
+        subject.perform(content_id)
+
+        assert_publishing_api_put_content(content_id, republish_matcher)
+        assert_publishing_api_patch_links(content_id)
+      end
+
+      it "does not publish the document" do
+        subject.perform(content_id)
+
+        expect(WebMock).not_to have_requested(:post, /publish/)
+      end
+
+      it "does not speak to rummager" do
+        subject.perform(content_id)
+
+        expect(WebMock).not_to have_requested(:post, /search/)
+      end
```

Comment:
would it be worthwhile to add "does not speak to email alert api" as a test as well?